assets/js/components/dashboard-sharing/UserRoleSelect.js

```diff
@@ -56,12 +60,25 @@ export default function UserRoleSelect( { moduleSlug } ) {
 	const sharedRoles = useSelect( ( select ) =>
 		select( CORE_MODULES ).getSharedRoles( moduleSlug )
 	);
+	const rolesChanged = useSelect( ( select ) =>
+		select( CORE_MODULES ).haveSharingSettingsChanged(
+			`${ moduleSlug }.sharedRoles`
+		)
+	);
 
 	useKeyCodesInside( [ ESCAPE ], wrapperRef, () => setEditMode( false ) );
 
 	const toggleEditMode = useCallback( () => {
+		if ( editMode ) {
+			trackEvent(
+				`${ viewContext }_sharing`,
+				'change_shared_roles',
+				moduleSlug
+			);
+		}
+
 		setEditMode( ! editMode );
-	}, [ editMode, setEditMode ] );
+	}, [ editMode, setEditMode, rolesChanged, viewContext, moduleSlug ] );
```

Comment:
Looks like this needs the `rolesChanged` variable removed:
```suggestion
	}, [ editMode, setEditMode, viewContext, moduleSlug ] );
```

Reply:
Hi @tofumatt, having taken a look into this I realised the changes didn't quite meet the spec. I have updated the code so it only tracks an event when the user has made changes to the roles between the point of entering edit mode and then leaving it. As a result, this dependency list has also been updated.